peer.go

```diff
@@ -456,6 +457,60 @@ func (p *Peer) sendRsp(rsp *Response) {
 	}
 }
 
+// callLocal invokes a method handler defined on p itself, rather than
+// forwarding it to the remote peer.
+func (p *Peer) callLocal(ctx context.Context, method string, data []byte) (*Response, error) {
+	handler, err := func() (Handler, error) {
+		p.μ.Lock()
+		defer p.μ.Unlock()
+		if err := p.err; err != nil {
+			return nil, err
+		}
+		handler, ok := p.imux[method]
+		if !ok {
+			const wildcardID = ""
+			handler, ok = p.imux[wildcardID]
+			if !ok {
+				return nil, errUnknownMethod{}
+			}
+		}
+		return handler, nil
+	}()
+	if rc, ok := err.(resultCoder); ok {
+		return nil, &CallError{Response: &Response{Code: rc.ResultCode()}}
+	} else if err != nil {
+		return nil, callError(err)
+	}
+
+	peerMetrics.callPending.Add(1)
+	defer peerMetrics.callPending.Add(-1)
+	result, err := handler(ctx, &Request{
+		Method: method,
+		Data:   data,
+	})
+	if err != nil {
+		// Propagate context cancellation without wrapping so the caller can
```

Comment:
Without wrapping? But you wrapped it?...

Reply:
Sorry, the comment is worded poorly. I mean without decorating it as an error from the server. I attempted to clarify, PTAL.